spamgourmet-clone/scripts/install-spamgourmet.sh

```diff
@@ -0,0 +1,275 @@
+
+	echo '##########################################################################'
+	echo '### link to exim4'
+	echo '##########################################################################'
+	cat <<-EOF >>/etc/exim4/conf.d/router/00_exim4-config_header
+	    spamgourmet_address:
+	      debug_print = "R: spamgourmet_address \$local_part@\$domain"
+	      driver = accept
+	      transport = sg_to_user_pipe
+	      domains = $DOMAIN
+	      errors_to = "[me]"
+
+	    outbound_spamgourmet_address:
+	      debug_print = "R: outbound_spamgourmet_address \$local_part@\$domain"
+	      driver = accept
+	      transport = sg_from_user_pipe
+	      domains = ob.$DOMAIN
+EOF
+
+	cat <<-EOF >>/etc/exim4/conf.d/transport/00_exim4-config_header
+	sg_to_user_pipe:
+	  debug_print = "T: sg_to_user_pipe for \$local_part@\$domain"
+	  driver = pipe
+	  return_output = false
+	  temp_errors = 69
+	  ignore_status = true
+	  environment = REPLYTO=\$reply_address:FROM=\$header_from::TO=\$header_to:
+	  command = /usr/local/lib/spamgourmet/mailhandler/spameater
+	  #user = "spamgourmet"
+	  #group = "spamgourmet"
+
+	sg_from_user_pipe:
+	  debug_print = "T: sg_from_user_pipe for \$local_part@\$domain"
+	  driver = pipe
+	  return_output = true
+	  environment = REPLYTO=\$reply_address:FROM=\$header_from:
+	  command = /usr/local/lib/spamgourmet/mailhandler/outbound
+	  # user = "spamgourmet"
+	  #group = "spamgourmet"
+EOF
+	# registration emails should not use spamgourmet website
+	sed -i "s/www.spamgourmet.com/$DOMAIN/g" /usr/local/lib/spamgourmet/modules/Mail/Spamgourmet/WebMessages.pm
+	# reply address masking uses # so let it pass
+	sed -i '/^CHECK_RCPT_/ s/\#//g' /etc/exim4/conf.d/main/01_exim4-config_listmacrosdefs
+	cd $SCRIPT_BASE_DIR
+	if [ -e ./dkim.private ]; then
+		echo '##########################################################################'
+		echo '### configure exim4 for dkim since there are dkim keys in the folder'
+		echo '### where this script is'
+		echo '##########################################################################'
+		mv dkim.* /etc/exim4/
+		chown root:Debian-exim /etc/exim4/dkim.*
+		chmod 640 /etc/exim4/dkim.*
+		cat <<-EOF >>/etc/exim4/conf.d/main/01_exim4-config_listmacrosdefs
+		#DKIM loading
+		DKIM_CANON = relaxed
+		DKIM_DOMAIN = \${sender_address_domain}
+		DKIM_PRIVATE_KEY = CONFDIR/dkim.private
+		DKIM_SELECTOR = $DKIM_SELECTOR
+EOF
+	else
+		echo '##########################################################################'
+		echo '### WARNING! CANNOT configure exim4 for dkim, missing dkim keys'
+		echo '##########################################################################'
+	fi
+	/var/lib/dehydrated/renewAllCerts.sh
+	service exim4 restart
+	service lighttpd restart
+}
+
+function configure_sg_db_connection_and_data {
+	#ensure that mysqld is running, - what about init systems?
+	#    mysqld_safe --no-watch
+	service mysql start
```

Comment:
Will the DB files be stored outside the Docker image?

Reply:
Eventually we must make that possible. Right now I don't care as long as I can get at least one test running.